q2_phylogenomics/tests/test_filter.py

```diff
@@ -0,0 +1,121 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import gzip
+import itertools
+import unittest
+
+from q2_types.per_sample_sequences import (
+    SingleLanePerSampleSingleEndFastqDirFmt,
+    SingleLanePerSamplePairedEndFastqDirFmt,
+    FastqGzFormat,
+)
+from qiime2 import Artifact
+from qiime2.plugin.testing import TestPluginBase
+
+
+class TestBowtie2Build(TestPluginBase):
+    package = 'q2_phylogenomics.tests'
+
+    # This test just makes sure this runs without error, which will include
+    # validating the contents.
+    def test_build(self):
+        genome = Artifact.load(self.get_data_path('sars2-genome.qza'))
+        self.plugin.methods['bowtie2_build'](genome)
+
+
+seq_ids_that_map = ['SARS2:6:73:941:1973#', 'SARS2:6:73:231:3321#',
+                    'SARS2:6:73:233:3421#', 'SARS2:6:73:552:2457#',
+                    'SARS2:6:73:567:7631#']
+seq_id_that_does_not_map = 'SARS2:6:73:356:9806#'
+
+
+class TestFilterSingle(TestPluginBase):
+    package = 'q2_phylogenomics.tests'
+
+    def setUp(self):
+        super().setUp()
+
+        self.demuxed_art = Artifact.load(self.get_data_path('single-end.qza'))
+        self.indexed_genome = Artifact.load(
+            self.get_data_path('sars2-indexed.qza'))
+
+    def test_filter_single_exclude_seqs(self):
+        obs_art, = self.plugin.methods['filter_single'](
+            self.demuxed_art, self.indexed_genome, exclude_seqs=True)
+        obs = obs_art.view(SingleLanePerSampleSingleEndFastqDirFmt)
+        obs_seqs = obs.sequences.iter_views(FastqGzFormat)
```

Comment:
I was hoping we had an skbio view type for this format, but apparently that's only on the fasta formats.